EnvVar for full functionality

svicario opened the pull request:
Adding EnvVar to match full functionality of phyloH. Some EnvVar would match both flag and value so that when option is not required the entire flag could be skiped.

svicario committed:
EnvVar for full functionality

Adding EnvVar to match full functionality of phyloH. Some EnvVar would match both flag and value so that when option is not required the entire flag could be skiped.

maricaantonacci reviewed (changes requested):
Hi Saverio,

the following changes might improve script readability:

<pre>
#!/bin/bash

# Branch of the phyloH repo - default to master
GITBRANCH=${GITBRANCH:-master}
LOG_FILE=${LOG_FILE:-run.log}
OUTPUT_PREFIX=${OUTPUT_PREFIX:-out}

EQUALQUANTITY=${$EQUALQUANTITY:-0}
PAIRSWISE=${$PAIRSWISE:-0}

# if GEOADD is set...
if [ ! -z ${GEOADD+x} ]; then
  OPTIONS="${OPTIONS} -G 1"
fi

# if TAXONOMY_FILENAME is set...
if [ ! -z ${TAXONOMY_FILENAME+x} ]; then
  OPTIONS="${OPTIONS} -t $TAXONOMY_FILENAME"
fi

# if PHYLOGENY_FILENAME is set...
if [ ! -z ${PHYLOGENY_FILENAME+x} ]; then
  OPTIONS="${OPTIONS} -f $PHYLOGENY_FILENAME"
fi

# if GROUP_FILENAME is set...
if [ ! -z "${GROUP_FILENAME+x}" ]; then
  OPTIONS="${OPTIONS} -g $GROUP_FILENAME"
fi

[...]

python phyloH/esecutorePhyloHPandas.py "${OPTIONS}" -s "$SAMPLE_FILENAME" -r "$Nrandomization" -o "$OUTPUT_PREFIX" -e "$EQUALQUANTITY" -k "$PAIRSWISE"
</pre>

svicario committed:
adding comments to code

added comments to code following good advice

svicario committed:
add option to run.sh

here the option relative to Geographic function and general interaction with GRASS are listed

svicario committed:
update run.sh

make name of number randomization nicer

svicario commented:
ecco la versione aggiornata
